src/hooks/use-force-max-amount.ts

```diff
@@ -0,0 +1,42 @@
+import { YieldDto } from "@stakekit/api-hooks";
+import { Maybe } from "purify-ts";
+import { useYieldBalance } from "./use-yield-balance";
+import { useMemo } from "react";
+
+/**
+ * Check if we need to use max amount for staking/unstaking
+ * based on yields requirements
+ */
+export const useForceMaxAmount = ({
+  type,
+  integration,
+}: {
+  type: "enter" | "exit";
+  integration: Maybe<YieldDto>;
+}) => {
+  const yieldBalance = useYieldBalance(integration);
+
+  const forceMax = useMemo(
+    () =>
+      integration
+        .chainNullable((v) =>
+          type === "enter"
+            ? v.args.enter.args?.amount
+            : v.args.exit?.args?.amount
+        )
+        .map((val) => val.minimum === -1 && val.maximum === -1)
+        .orDefault(false),
+    [integration, type]
+  );
+
+  const canStake =
+    type === "enter" && (!forceMax || !yieldBalance.data?.length);
```

Comment:
in order to stake - if we have max/min -1, does it make sense to just check if yield balances are empty, or we need to check if there are no `available` or `staked` balances? 
@Philippoes

Reply:
Yeah, if the user has a yield balance with Tezos we could technically gray out the stake button or not allow them to continue. (We do have the guard in the backend that if you try to enter, but have yield balances, it throws an error).